account-kit/core/src/store/store.ts

```diff
@@ -311,3 +341,29 @@ export const createEmptySmartAccountClientState = (chains: Chain[]) => {
     return acc;
   }, {} as StoreState["smartAccountClients"]);
 };
+
+/**
+ * Reads and removes the specified query params from the URL.
+ *
+ * @param {T} keys object whose values are the query parameter keys to read and
+ * remove
+ * @returns {{ [K in keyof T]: string | undefined }} object with the same keys
+ * as the input whose values are the values of the query params.
+ */
+function takeQueryParams<T extends Record<string, string>>(
```

Comment:
nit the function name at first felt like a read only function, should we make this more explicit? We split it into two functions? Rename it to getAndUpdateQueryUrlParam?

Not going to block on this btw.

Reply:
We also have a very similar function in account kit signer. Should we consolidate somewhere?

Reply:
if we consolidate, probs best to put it in signer since it's pretty signer specific

Reply:
You're right, I should make the name more clear. I was following the Rust naming convention of take meaning to remove the current value and return it, e.g. [Option::take](https://doc.rust-lang.org/std/option/enum.Option.html#method.take), but that's unlikely to be familiar to readers here.

Reply:
Just pushed a version that consolidates all query param reading into the signer constructor.